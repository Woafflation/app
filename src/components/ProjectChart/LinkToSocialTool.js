import React, { Fragment } from 'react'
import { Link } from 'react-router-dom'

export const LinkToSocialTool = ({ slug, ticker, name }) => {
  const social =
    window.encodeURIComponent(
      `${slug || ''}${ticker ? ' OR ' + ticker : ''}${
        name ? ' OR ' + name : ''
      }`
    ) + `?asset=${slug}`
  return (
    <Fragment>
      {slug ? (
        <Link to={`/labs/trends/explore/${social}`}>See Social Trends</Link>
      ) : (
        <span>...</span>
      )}
    </Fragment>
  )
}

export default LinkToSocialTool
