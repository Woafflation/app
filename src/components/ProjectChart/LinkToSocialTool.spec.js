/* eslint-env jest */
import React from 'react'
import LinkToSocialTool from './LinkToSocialTool'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe('LinkToSocialTool', () => {
  it('Smoke', () => {
    const project = {
      name: 'Santiment Network Token',
      ticker: 'SAN',
      slug: 'santiment'
    }
    const wrapper = shallow(<LinkToSocialTool {...project} />)
    expect(wrapper.find('Link').props().to).toEqual(
      '/labs/trends/explore/santiment%20OR%20SAN%20OR%20Santiment%20Network%20Token?asset=santiment'
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it("LinkToSocialTool, if project doesn't have name", () => {
    const project = {
      name: null,
      ticker: 'SAN',
      slug: 'santiment'
    }
    const wrapper = shallow(<LinkToSocialTool {...project} />)
    expect(wrapper.find('Link').props().to).toEqual(
      '/labs/trends/explore/santiment%20OR%20SAN?asset=santiment'
    )

    const projectWithoutTicker = {
      name: null,
      ticker: null,
      slug: 'santiment'
    }
    const wrapper2 = shallow(<LinkToSocialTool {...projectWithoutTicker} />)
    expect(wrapper2.find('Link').props().to).toEqual(
      '/labs/trends/explore/santiment?asset=santiment'
    )
  })
})
