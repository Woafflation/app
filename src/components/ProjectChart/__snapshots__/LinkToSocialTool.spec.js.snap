// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToSocialTool Smoke 1`] = `
<Fragment>
  <Link
    replace={false}
    to="/labs/trends/explore/santiment%20OR%20SAN%20OR%20Santiment%20Network%20Token?asset=santiment"
  >
    See Social Trends
  </Link>
</Fragment>
`;
